server/db_models/pluser.py

```diff
@@ -1,13 +1,42 @@
+import re
 from app import db
+from app import bcrypt
+from sqlalchemy.orm import validates
 
 # Mock User class for testing.
 class plUser(db.Model):
     __tablename__ = "plUser" # use CamelCase for table names
     id = db.Column(db.Integer, primary_key=True)
-    username = db.Column(db.String(64), index=True, unique=True)
-    login_email = db.Column(db.String(120), index=True, unique=True)
-    password_hash = db.Column(db.String(128))
-    location = db.Column(db.String(64))
+    username = db.Column(db.String(64), index=True, unique=True, nullable=False)
+    login_email = db.Column(db.String(64), index=True, unique=True, nullable=False)
+    password_hash = db.Column(db.String(128), nullable=False)
+    location = db.Column(db.String(64), nullable=True)
 
     def __repr__(self):
-        return '<plUser {}>'.format(self.username) 
+        return '<plUser {}>'.format(self.username)
+    
+    def set_password(self, password):
+        if not password:
+            raise AssertionError('No password provided')
+        if len(password) < 6 or len(password) > 64:
+            raise AssertionError('Password must be between 6 and 64 characters')
+
+        hash = bcrypt.generate_password_hash(password.encode('utf-8'))
+        self.password_hash = hash.decode('utf-8')
+
```

Comment:
A small issue I have is that when you register, the register_handler does a validation, and the db does a validation. I'm wondering what's the purpose of having two?

Reply:
The main purpose was to ensure data integrity for any future uses of the user model outside of the register_handler. On that note, we should probably add some validation for other fields in a future PR as the model validations I added were not comprehensive.